### ISSUE TITLE
Top issues to tackle

### ISSUE THREAD

**Issue by AdamLearns** [AdamLearns/AnimalRoyale#1]
Yes, I should be filing separate issues for all of these, but I don't want to spend the time working on this. 😓

* Drowning doesn't have a special message and doesn't strike you out on the scoreboard  
  * Teleporting shouldn't have a cooldown at the beginning of the game. Maybe there's a "random" command to relocate you  
    * Change the "you win" message in Twitch chat to show kill count  
      * This is this message:  
        * `AdamLearnsBot: @darkstoorm won the battle! PogChamp GG, everyone!`
        * It should print the top few players by kill count
    * Need to make the game more user-friendly  
      * Allow for something like "\!tnt left"  
      * Try to get the lobby to start sooner. I think it's waiting for the tickrate to settle down, but perhaps we can speed that process up somehow
      * Show a timer for the lobby
      * Show a timer for how long the game has been going  
        * Indicate when sudden death will happen?  
      * Show a player counter  
        * How many total vs. how many are left  
      * Show TNT cooldown timer

**Comment by kevinrpb** [comment]
I was looking into how to show the timers and such on-screen. I came across this plugin: https://github.com/toxicity188/BetterHud to add custom HUDs server-side. Maybe we can use it or at least take it as reference of how to do it, because I wasn't able to find much info in PaperMC docs.

**Comment by AdamLearns** [comment]
> I was looking into how to show the timers and such on-screen. I came across this plugin: [toxicity188/BetterHud](https://github.com/toxicity188/BetterHud) to add custom HUDs server-side.

Just to be clear, is that _only_ server-side? I looked through their documentation and it appears to be, but I'm all over the place today. 👀

I'd be fine with taking this on as a dependency if there's no other obvious way to do this stuff. 👍

**Comment by kevinrpb** [comment]
> Just to be clear, is that only server-side?

AFAICT yes, but I still have to properly check out how it works and try it out. I would also like to peer through their source to see what are the APIs that allow them to do this from the server. If they are not too cumbersome, we may be able to do something simple ourselves.

**Comment by kevinrpb** [comment]
Wow so I took a look at how these custom HUDs are created and it's pretty wild lol. Basically people abuse the game's ResourcePacks and use custom font definitions for un-set unicode characters. Then assign those unicode characters to custom textures.

So basically all these UIs are just text... 🤣 Found this video explaining it:

[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/EL2X6ppZSCQ/0.jpg)](https://www.youtube.com/watch?v=EL2X6ppZSCQ)

**Comment by kevinrpb** [comment]
@DarkStoorM so you can TP before the game starts (it won't make you take dmg until round starts).

What Adam suggested is making a `!relocate` command that you can use at the beginning to get yourself elsewhere.

Also I was thinking about how we 'ban' biomes with lots of water. Maybe we could generate a world that doesn't have those biomes at all?

**Comment by AdamLearns** [comment]
> Also I was thinking about how we 'ban' biomes with lots of water. Maybe we could generate a world that doesn't have those biomes at all?

I like the idea of water for two reasons:

- When attacking someone who is on water, you just need to knock them into the water rather than have a direct hit
- When _you're_ on the water, it's harder for someone to slide a TNT into you since it'll sink

So I'd like to see how `!relocate` does (or something similar). We could also potentially create bigger platforms when we spawn someone on the water (e.g. 10x10 instead of 1x1).

**Comment by kevinrpb** [comment]
> I like the idea of water for two reasons

Yeah, I also like having some water. What I was suggesting is that we don't generate water-only or water-mostly biomes. This should also make it easier for the plugin to find viable biomes.
